.github/workflows/yocto-build-deploy.yml

```diff
@@ -0,0 +1,639 @@
+        required: false
+        type: boolean
+        default: true
+      deploy-ami:
+        description: Whether to deploy an AMI to AWS
+        required: false
+        type: boolean
+        default: false
+      sign-image:
+        description: Whether to sign image for secure boot
+        required: false
+        type: boolean
+        default: false
+      os-dev:
+        description: Enable OS development features
+        required: false
+        type: boolean
+        default: false
+      deploy-esr:
+        description: Enable to deploy ESR
+        required: false
+        type: boolean
+        default: false
+
+jobs:
+  build:
+    name: Build
+    runs-on: ${{ fromJSON(inputs.runs-on) }}
+    environment: ${{ inputs.github-environment }}
+
+    env:
+      MACHINE: "${{ inputs.machine }}"
+      automation_dir: "${{ github.workspace }}/balena-yocto-scripts/automation"
+      BALENARC_BALENA_URL: "${{ inputs.balena-environment }}"
+      WORKSPACE: ${{ github.workspace }}
+      VERBOSE: verbose
+      YOCTO_CACHE_DIR: /var/lib/yocto
+      BARYS_ARGUMENTS_VAR: ""
+
+    outputs:
+      hostapp_path: ${{ steps.build.outputs.hostapp_path }}
+      os_version: ${{ steps.build-lib.outputs.os_version }}
+      device_slug: ${{ steps.build-lib.outputs.device_slug }}
+      deploy_artifact: ${{ steps.build-lib.outputs.deploy_artifact }}
+      is_private: ${{ steps.build-lib.outputs.is_private }}
+      dt_arch: ${{ steps.build-lib.outputs.dt_arch }}
+      meta_balena_version: ${{ steps.build-lib.outputs.meta_balena_version }}
+      yocto_scripts_ref: ${{ steps.build-lib.outputs.yocto_scripts_ref }}
+
+    defaults:
+      run:
+        working-directory: .
+        shell: bash --noprofile --norc -eo pipefail -x {0}
+
+    steps:
+      # https://github.com/actions/checkout
+      - name: Clone device repository
+        uses: actions/checkout@v3
+        with:
+          repository: ${{ inputs.device-repo }}
+          token: ${{ secrets.GITHUB_TOKEN }}
+          ref: ${{ inputs.device-repo-ref }}
+          submodules: true
+
+      - name: Update meta-balena submodule to ${{ inputs.meta-balena-ref }}
+        if: inputs.meta-balena-ref != ''
+        working-directory: ./layers/meta-balena
+        run: |
+          git config --add remote.origin.fetch '+refs/pull/*:refs/remotes/origin/pr/*'
+          git fetch --all
+          git checkout --force "${{ inputs.meta-balena-ref }}"
+          git submodule update --init --recursive
+
+      - name: Update balena-yocto-scripts submodule to ${{ inputs.yocto-scripts-ref }}
+        if: inputs.yocto-scripts-ref != ''
+        working-directory: ./balena-yocto-scripts
+        run: |
+          git config --add remote.origin.fetch '+refs/pull/*:refs/remotes/origin/pr/*'
+          git fetch --all
+          git checkout --force "${{ inputs.yocto-scripts-ref }}"
+          git submodule update --init --recursive
+
+      - name: Device repository check
```

Comment:
This should come before updating meta-balena and balena-yocto-scripts as those will only exist on a device repository.

Reply:
Good call, I'll make the change.